src/arrayMethodJoin.js

```diff
@@ -4,9 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let arr = '';
+    const separatorCopy = separator + '';
+    // console.log(this);
+
+    for (let i = 0; i < this.length; i++) {
+      arr += (this[i] === undefined || this[i] === null ? '' : this[i])
+      + (i < this.length - 1 ? separatorCopy : '');
```

Comment:
here also, it's better not to do calculations within calculations 🤯. Better use variables. You certainly won a few lines, but it turned into a mess 😄